clang/test/CodeGenSYCL/debug-info-srcpos-kernel.cpp

```diff
@@ -22,8 +21,8 @@ int main() {
   return 0;
 }
 
-// CHECK-OLD: define{{.*}} spir_kernel {{.*}}void @_ZTSZ4mainE15kernel_function(i32*{{.*}}){{.*}} !dbg [[KERNEL:![0-9]+]] {{.*}}{
-// CHECK-NEW: define{{.*}} spir_kernel {{.*}}void @_ZTSZ4mainE15kernel_function(i32 addrspace(4)*{{.*}}){{.*}} !dbg [[KERNEL:![0-9]+]] {{.*}}{
+
+// CHECK: define{{.*}} spir_kernel {{.*}}void @_ZTSZ4mainE15kernel_function(i32 addrspace(1)*{{.*}}){{.*}} !dbg [[KERNEL:![0-9]+]] {{.*}}{
```

Comment:
Interesting.
@asavonic, this looks like a bug with address space handling.
1. Note, clang emit OpenCL kernel with "generic pointer" argument. 
2. Clang does not emit warnings about captured pointers.

I think both these issues should be fixed for the cases when USM feature is not available.